DatadogCore/Tests/Datadog/Core/Persistence/Files/DirectoryTests.swift

```diff
@@ -171,6 +171,156 @@ class DirectoryTests: XCTestCase {
+        XCTAssertNotNil(modifiedFile)
+        XCTAssertEqual(file3.url, modifiedFile?.url)
+    }
+
+    func testLatestModifiedFile_whenDirectoryContainsSubdirectories_returnsLatestModifiedFile() throws {
+        let directory = try Directory(withSubdirectoryPath: uniqueSubdirectoryName())
+        defer { directory.delete() }
+
+        let subdirectory1 = try directory.createSubdirectory(path: "subdirectory1")
+        let subdirectory2 = try directory.createSubdirectory(path: "subdirectory2")
+        let subdirectory3 = try directory.createSubdirectory(path: "subdirectory3")
+
+        let file1 = try subdirectory1.createFile(named: "file1")
+        let file2 = try subdirectory2.createFile(named: "file2")
+        let file3 = try subdirectory3.createFile(named: "file3")
+
+        try file1.append(data: .mock(ofSize: 1))
+        try file2.append(data: .mock(ofSize: 2))
+        try file3.append(data: .mock(ofSize: 3))
+
+        let modifiedFile = try directory.mostRecentModifiedFile(before: .init())
+        XCTAssertNotNil(modifiedFile)
+        XCTAssertEqual(file3.url, modifiedFile?.url)
+    }
+
+    func testLatestModifiedFile_whenDirectoryContainsFilesAndSubdirectories_returnsLatestModifiedFile() throws {
+        let directory = try Directory(withSubdirectoryPath: uniqueSubdirectoryName())
+        defer { directory.delete() }
+
+        let subdirectory1 = try directory.createSubdirectory(path: "subdirectory1")
+        let subdirectory2 = try directory.createSubdirectory(path: "subdirectory2")
+        let subdirectory3 = try directory.createSubdirectory(path: "subdirectory3")
+
+        let file1 = try subdirectory1.createFile(named: "file1")
+        let file2 = try subdirectory2.createFile(named: "file2")
+        let file3 = try subdirectory3.createFile(named: "file3")
+
+        try file1.append(data: .mock(ofSize: 1))
+        try file2.append(data: .mock(ofSize: 2))
+        try file3.append(data: .mock(ofSize: 3))
+
+        let modifiedFile = try directory.mostRecentModifiedFile(before: .init())
+        XCTAssertNotNil(modifiedFile)
+        XCTAssertEqual(file3.url, modifiedFile?.url)
+    }
+
+    func testLatestModifiedFile_whenDirectoryContainsFilesAndFileIsDeleted_returnsLatestModifiedFile() throws {
+        let directory = try Directory(withSubdirectoryPath: uniqueSubdirectoryName())
+        defer { directory.delete() }
+
+        let file1 = try directory.createFile(named: "file1")
+        let file2 = try directory.createFile(named: "file2")
+        let file3 = try directory.createFile(named: "file3")
+
+        try file1.append(data: .mock(ofSize: 1))
+        try file2.append(data: .mock(ofSize: 2))
+        try file3.append(data: .mock(ofSize: 3))
+
+        try file2.delete()
+
+        let modifiedFile = try directory.mostRecentModifiedFile(before: .init())
+        XCTAssertNotNil(modifiedFile)
+        XCTAssertEqual(directory.url, modifiedFile?.url)
```

Comment:
nit: should it also compare `modifiedFile` and `file3` urls?

Reply:
Oh right, so it returns the directory itself? So for watchdog termination, it doesn't matter if we get the timestamp of the last modified file or the directory itself?

Reply:
correct, what matters is it has to be lower than the given date, because we don't want to take any updated dir/file which was affect of current session.

Reply:
Got it, thanks for the explanation.